runtime/jcl/common/criusupport.c

```diff
@@ -107,6 +107,20 @@ Java_org_eclipse_openj9_criu_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, j
 	return res;
 }
 
+jboolean JNICALL
+Java_org_eclipse_openj9_criu_CRIUSupport_isCheckpointPhase(JNIEnv *env, jclass unused)
+{
+	J9VMThread *currentThread = (J9VMThread *) env;
+	J9JavaVM *vm = currentThread->javaVM;
+	jboolean res = JNI_FALSE;
+
+	if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_CRIU_CHECKPOINT_PHASE)) {
+		res = JNI_TRUE;
+	}
+
+	return res;
+}
```

Comment:
Using a runtimeFlags bit provides a very limited view on what's happened.  I would suggest adding a struct that captures some relevant state at the time of the snapshot and stores it in the J9JavaVM.

Things to capture would include whether to allow subsequent snapshots, might be the time / date of the snapshot, any additional state to identify the env used for the snapshot.  It doesn't have to capture all this right away, but it provides an easy place to get the info.

Then this function can be:
```c
boolean isCheckpointAllowed() {
	J9VMThread *currentThread = (J9VMThread *) env;
	J9JavaVM *vm = currentThread->javaVM;

	if (isCRIUSupportEnabled()) {
		if (vm->snapshotState == NULL) {
			return true;
		} else {
			return vm->snapshotState->checkpointAllowed;
		}
	}
	return false;
}
```